src/bin/mosaic-start.ts

```diff
@@ -45,8 +47,17 @@ mosaic
       // reuse params from node start command
       graphDescription.mosaicDir = mosaicDir;
       graphDescription.ethereumRpcPort = rpcPort;
-      const graph = new Graph(graphDescription);
-      graph.start();
+
+      new Graph(graphDescription).start().then(function () {
+        let subGraphDeployer;
+        // options.origin passed only in case of starting an auxiliary chain
+        if (options.origin) {
+          subGraphDeployer = new SubGraphDeployer(graphDescription, options.origin, chain);
+        } else {
+          subGraphDeployer = new SubGraphDeployer(graphDescription, chain, null);
```

Comment:
If I run `./mosaic start 1407` the else option will execute, right? 
If yes, the `chain` will be `1407`?   But the second argument of subgraph constructor is originChain?

Reply:
When we want to start 1407, we would call `./mosaic start 1407 --origin ropsten`. This would lead to execution of if block. 
Also as we discussed, with the ticket where we are realigning the directory structures we would add strict validations which would lead to `./mosaic start 1407` fail.

Reply:
I am not 100% clear about this. My point was are we passing the auxilairy chain identifier instead of origin chain identifier in the constructor of `SubGraphDeployer`.

Reply:
Nope. 

While starting `./mosaic start 1407 --origin ropsten`
we call `subGraphDeployer = new SubGraphDeployer(graphDescription, 'ropsten', 1407);`

And while starting `./mosaic start ropsten`
we call `subGraphDeployer = new SubGraphDeployer(graphDescription, 'ropsten');`

Reply:
and for case `./mosaic start 1407` 
```
subGraphDeployer = new SubGraphDeployer(graphDescription, '1407', null);
```

right?